src/Redis/Orleans.Reminders.Redis/Providers/RedisReminderTableOptions.cs

```diff
@@ -10,29 +11,52 @@ namespace Orleans.Configuration
     /// </summary>
     public class RedisReminderTableOptions
     {
-
         /// <summary>
-        /// The connection string.
+        /// Gets or sets the Redis client options.
         /// </summary>
-        public string ConnectionString { get; set; } = "localhost:6379";
+        [RedactRedisConfigurationOptions]
+        public ConfigurationOptions ConfigurationOptions { get; set; }
 
         /// <summary>
-        /// The database number.
+        /// The delegate used to create a Redis connection multiplexer.
         /// </summary>
-        public int? DatabaseNumber { get; set; }
+        public Func<RedisReminderTableOptions, Task<IConnectionMultiplexer>> CreateMultiplexer { get; set; } = DefaultCreateMultiplexer;
 
         /// <summary>
-        /// The delegate used to create a Redis connection multiplexer.
+        /// Entry expiry, null by default. A value should be set ONLY for ephemeral environments (like in tests).
+        /// Setting a value different from null will cause reminder entries to be deleted after some period of time.
         /// </summary>
-        public Func<RedisReminderTableOptions, Task<IConnectionMultiplexer>> CreateMultiplexer { get; set; } = DefaultCreateMultiplexer;
+        public TimeSpan? EntryExpiry { get; set; } = null;
 
         /// <summary>
         /// The default multiplexer creation delegate.
         /// </summary>
-        public static async Task<IConnectionMultiplexer> DefaultCreateMultiplexer(RedisReminderTableOptions options)
+        public static async Task<IConnectionMultiplexer> DefaultCreateMultiplexer(RedisReminderTableOptions options) => await ConnectionMultiplexer.ConnectAsync(options.ConfigurationOptions);
+    }
+
+    internal class RedactRedisConfigurationOptions : RedactAttribute
+    {
+        public override string Redact(object value) => value is ConfigurationOptions cfg ? cfg.ToString(includePassword: false) : base.Redact(value);
+    }
+
+    /// <summary>
+    /// Configuration validator for <see cref="RedisReminderTableOptions"/>.
+    /// </summary>
+    public class RedisReminderTableOptionsValidator : IConfigurationValidator
+    {
+        private readonly RedisReminderTableOptions _options;
+
+        public RedisReminderTableOptionsValidator(RedisReminderTableOptions options)
         {
-            return await ConnectionMultiplexer.ConnectAsync(options.ConnectionString);
+            _options = options;
         }
 
+        public void ValidateConfiguration()
+        {
+            if (_options.ConfigurationOptions == null)
+            {
+                throw new OrleansConfigurationException($"Invalid {nameof(RedisReminderTableOptions)} values for {nameof(RedisReminderTable)}. {nameof(_options.ConfigurationOptions)} is required.");
```

Comment:
nit space?

Reply:
It's a new sentence

Reply:
I clarified the wording

Reply:
My bad; I thought it was a qualified type!